3rdparty/openssl/CMakeLists.txt

```diff
@@ -129,146 +276,169 @@ ExternalProject_Add(
 # find the former.
+
+set(OPENSSL_ASSEMBLY_SCRIPTS
+    crypto/aes/asm/aesni-mb-x86_64
+    crypto/aes/asm/aesni-sha1-x86_64
+    crypto/aes/asm/aesni-sha256-x86_64
+    crypto/bn/asm/rsaz-avx2
+    crypto/bn/asm/rsaz-x86_64
+    crypto/bn/asm/x86_64-gf2m
+    crypto/bn/asm/x86_64-mont
+    crypto/bn/asm/x86_64-mont5
+    crypto/ec/asm/ecp_nistz256-x86_64
+    crypto/ec/asm/x25519-x86_64
+    crypto/md5/asm/md5-x86_64
+    crypto/modes/asm/aesni-gcm-x86_64
+    crypto/modes/asm/ghash-x86_64
+    crypto/sha/asm/keccak1600-x86_64
+    crypto/sha/asm/sha1-mb-x86_64
+    crypto/sha/asm/sha1-x86_64
+    crypto/sha/asm/sha256-mb-x86_64
+    crypto/sha/asm/sha512-x86_64)
+
+set(OPENSSL_3_ASSEMBLY_SCRIPTS
+    crypto/aes/asm/aes-x86_64 crypto/aes/asm/bsaes-x86_64
+    crypto/bn/asm/rsaz-avx512)
+
+set(OPENSSL_GENERATED_COMMANDS
+    COMMAND ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_BUILD_GENERATED}/crypto/perlasm COMMAND ${CMAKE_COMMAND} -E copy
+    ${INTELSSL_DIR}/x86_64-xlate.pl
+    ${OPENSSL_BUILD_GENERATED}/crypto/perlasm/x86_64-xlate.pl COMMAND
+    ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_BUILD_GENERATED}/crypto/aes/asm COMMAND ${CMAKE_COMMAND} -E
+    make_directory ${OPENSSL_BUILD_GENERATED}/crypto/bn/asm COMMAND
+    ${CMAKE_COMMAND} -E make_directory ${OPENSSL_BUILD_GENERATED}/crypto/ec/asm
+    COMMAND ${CMAKE_COMMAND} -E copy
+    ${OPENSSL_DIR}/crypto/ec/ecp_nistz256_table.c
+    ${OPENSSL_BUILD_GENERATED}/crypto/ec/ecp_nistz256_table.c COMMAND
+    ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_BUILD_GENERATED}/crypto/md5/asm COMMAND ${CMAKE_COMMAND} -E
+    make_directory ${OPENSSL_BUILD_GENERATED}/crypto/modes/asm COMMAND
+    ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_BUILD_GENERATED}/crypto/sha/asm)
+
+set(OPENSSL_3_GENERATED_COMMANDS
+    COMMAND ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/perlasm COMMAND ${CMAKE_COMMAND} -E
+    copy ${OPENSSL_3_DIR}/crypto/perlasm/x86_64-xlate.pl
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/perlasm/x86_64-xlate.pl COMMAND
+    ${CMAKE_COMMAND} -E copy ${OPENSSL_3_DIR}/crypto/perlasm/x86_64-support.pl
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/perlasm/x86_64-support.pl COMMAND
+    ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/aes/asm COMMAND ${CMAKE_COMMAND} -E
+    make_directory ${OPENSSL_3_BUILD_GENERATED}/crypto/bn/asm COMMAND
+    ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/ec/asm COMMAND ${CMAKE_COMMAND} -E copy
+    ${OPENSSL_3_DIR}/crypto/ec/ecp_nistz256_table.c
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/ec/ecp_nistz256_table.c COMMAND
+    ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/md5/asm COMMAND ${CMAKE_COMMAND} -E
+    make_directory ${OPENSSL_3_BUILD_GENERATED}/crypto/modes/asm COMMAND
+    ${CMAKE_COMMAND} -E make_directory
+    ${OPENSSL_3_BUILD_GENERATED}/crypto/sha/asm)
+
+foreach (source_file ${OPENSSL_ASSEMBLY_SCRIPTS})
+  list(
```

Comment:
Can also make a macro for this

Reply:
Made a macro, but I didn't realize SCRIPT was a key word in CMake until reviewing the change on github. Will leave this open to remind me to change the variable name for clarity.